src/lib/logstore/log_store.cpp

```diff
@@ -405,7 +408,7 @@ void HomeLogStore::flush_sync(logstore_seq_num_t upto_seq_num) {
     if (upto_seq_num == invalid_lsn()) { upto_seq_num = m_records.active_upto(); }
 
     // if we have flushed already, we are done
-    if (!m_records.status(upto_seq_num).is_active) { return; }
+    if (m_records.status(upto_seq_num).is_completed) { return; }
```

Comment:
This status is a tri-state check.
1) Inactive: We have not received this lsn at all
2) Active: Received LSN, yet to flush
3) Completed: Completed flush.

So we only need to flush if it is not active. This change will flush when it is inactive as well. So I don't think we should make this change.

Reply:
I get the point, we need to flush only active, it make sense.
However  I am not getting how `is_active == false` implies  log getting flushed?  are we reset bit in `m_active_slot_bits` when doing update?  I didnt find this part of code in SISL.